plugins/dialog/plugin.js

```diff
@@ -300,6 +300,41 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 			definition: definition
 		}, editor ).definition;
 
+		var me = this,
+			contents = this.definition.contents;
+
+		// Overwrite definition with default values (#2277).
+		for ( var key in contents ) {
+			var contentObj = contents[ key ];
+			CKEDITOR.tools.array.forEach( contentObj.elements, function( element ) {
+				setDefaultValues( me, contentObj.id, element );
+			} );
+		}
+
+		function setDefaultValues( dialog, tabId, element ) {
+			var dialogId = dialog._.name;
+			if ( element.type === 'vbox' || element.type === 'hbox' ) {
+				return CKEDITOR.tools.array.forEach( element.children, function( child ) {
+					setDefaultValues( dialog, tabId, child );
+				} );
+			}
+
+			var defaultValue = getFieldConfig( dialog._.editor, dialogId, tabId, element.id );
+			if ( defaultValue ) {
+				element[ 'default' ] = defaultValue;
+			}
+		}
+
+		function getFieldConfig( editor, dialogId, tabId, fieldName ) {
+			var defaultValues = editor.config.dialog_defaultValues || CKEDITOR.config.dialog_defaultValues;
```

Comment:
The global `CKEDITOR.config` should be considered here. Only per-instance config applies here.

Global config should only be considered when creating `config` for a editor instance.

Imagine that someone _don't want any_ custom default values to be applied for a particular instance. While the default `CKEDITOR.config.dialog_defaultValues` contained something. I'd do it that way:

```javascript
CKEDITOR.config.dialog_defaultValues = {
	// values here
};
// (…)
CKEDITOR.replace( 'my-editor', {
	dialog_defaultValues: false
} );
```

The above code would be handled incorrectly with this approach.